### DISCUSSION TITLE
Panel data Regression

### DISCUSSION THREAD

**Discussion by rasooltaban** [alan-turing-institute/sktime#6875] [Q&A]
Hi all,

I’m working on a project involving panel data with thousands of observations from stocks across various countries. For each stock, I have multivariate time-series data, including open price, close price, volume, demographic information, and so on. My goal is to forecast the next day's close price for each stock, implying the target variable is itself a time series.

I’ve been exploring the sktime documentation but found limited guidance on how to handle panel data regression. Based on the package's assumptions, as outlined here **https://www.sktime.net/en/latest/examples/02_classification.html**, it seems that the package assumes the target variable to be a single point for each observation in panel data. 
Could anyone help with this matter?

Also, I have tried to use the "ComposableTimeSeriesForestRegressor" function, but I’m struggling with the pipeline definition and couldn’t find illustrative examples. Has anyone used this function before? If so, could you share an example or point me towards relevant documentation?

Thank you in advance for your support.

**Comment by fkiraly** [comment]
@rasooltaban, I think you have misdiagnosed your problem but not your data.

You are indeed dealing with panel data, but with forecasting, not time series classification. Because, as you say, the target is itself a time series, and not a single observation, as TSC assumes.

You would select the close price as endogenous variable `y`, and for the remaining variables, you can decide whether you want to treat them as exogeneous or endogeneous.

The vignette that applies is the forecasting fit/predict as in the examples notebook 01 - specifically from section 1.2.6, "panel forecasts".

Please let us know if you need forther advice!

**Reply by rasooltaban** [reply]
Hi @fkiraly,
Thank you for the quick reply.

It is unclear to me what you mean by: “dealing with forecasting, not time series classification”. In the context of panel data regression, I understand that we are dealing with a forecasting problem where the time horizon is 1-period ahead.
Could you please confirm if, in your approach and Panel data Regression formulation, the target variable y is treated as a continuous value, where each panel data observation (which includes multivariate time-series data) corresponds to a single forecasted value? If so, how should we handle cases where the target variable itself is a time series?

The example you have mentioned (Notebook 01-section 1.2.6) shows an ARIMA model, but I would like to apply the Random Forest Regressor to this time series problem. To do that I want to apply "**ComposableTimeSeriesForestRegressor**" function. Would you be able to share an example or point me toward relevant documentation?

**Reply by fkiraly** [reply]
> Could you please confirm if, in your approach and Panel data Regression formulation, the target variable y is treated as a continuous value, where each panel data observation (which includes multivariate time-series data) corresponds to a single forecasted value? If so, how should we handle cases where the target variable itself is a time series?

I think there is a terminology and an interface issue.

* "time series regression" - often also called "time series extrinsic regression", is predicting a single float value per time series. When seeing "time series regression" in the sktime docs, this is what is meant.
* "time series forecasting" - a subcase of "time series intrinsic regression", is predicting a series of float values on the same axis as the original time series. In sktime, these tasks fall under "forecasting".

To apply a random forest regressor to the time series problem, use `make_reduction` or `DirectReductionForecaster`, and the `sklearn` (or an sklearn compatible) random forest, not the "time series regression" estimator, which does not do forecasting (which is what you want)
